features/FEATURE_BLE/ble/pal/PalGattClient.h

```diff
@@ -455,13 +455,31 @@ class GattClient {
      * @param connection_handle The handle of the connection to send this request to.
      * @param attribute_handle Handle of the attribute to write.
      * @param value The value to write.
+     * @param sign_counter Signed write counter to use for this command.
      *
      * @return BLE_ERROR_NONE or an appropriate error.
      */
     virtual ble_error_t signed_write_without_response(
         connection_handle_t connection_handle,
         attribute_handle_t characteristic_value_handle,
-        const ArrayView<const uint8_t>& value
+        const ArrayView<const uint8_t>& value,
+        uint32_t sign_counter
+    ) = 0;
+
+    /**
+     * Set the signing counter associated with the peer. This counter will be used
+     * to verify the signed messages coming from that peer.
+     *
+     * @note see BLUETOOTH SPECIFICATION Version 5.0 | Vol 3, Part H Section 2.4.5
+     *
+     * @param connection_handle The handle of the connection for the peer.
+     * @param sign_counter Counter used to verify incoming signed messages.
+     *
+     * @return BLE_ERROR_NONE or an appropriate error.
+     */
+    virtual ble_error_t set_peer_signing_counter(
```

Comment:
The GattClient doesn't handle signed messages coming from a peer; the GattServer handles that.

Reply:
doh